src/org/sosy_lab/java_smt/solvers/z3/Z3FormulaCreator.java

```diff
@@ -970,7 +971,7 @@ private FloatingPointNumber convertFloatingPoint(FloatingPointType pType, Long p
           expo,
           mant,
           pType.getExponentSize(),
-          pType.getMantissaSize());
+          pType.getMantissaSizeWithSignBit());
```

Comment:
This seems wrong and there are several more such cases in this class.

Reply:
Even with the current changes, this is still a behavioral change, and it is unclear whether it is a bug fix or breaks something.

I think all changes in `Z3FormulaCreator` need to be checked carefully against the Z3 docs to make sure that they really match.